### PR TITLE
python3Packages.astropy: apply patch removing the usage of polyfill.io

### PR DESCRIPTION
**Pull request by LeSuisse**
## Description of changes

`cdn.polyfill.io` is now considered to be a bad actor.

https://sansec.io/research/polyfill-supply-chain-attack

Disabled on more test that was flaky on my side.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/astropy/default.nix [hunk 1]
```diff
@@ -83,6 +92,8 @@ buildPythonPackage rec {
     "test_sidereal_lon_independent"
     "test_timedelta_full_precision_arithmetic"
     "test_datetime_to_timedelta"
+
+    "test_datetime_difference_agrees_with_timedelta_no_hypothesis"
```
Comment (doronbehar) [inline pkgs/development/python-modules/astropy/default.nix:96]: How is this related to the malware? The source code for the test doesn't seem to be related to the HTML template:

https://github.com/astropy/astropy/blob/v6.0.1/astropy/time/tests/test_precision.py#L759

The package seem to build fine as of today:

https://hydra.nixos.org/build/263062097

**Review by doronbehar — COMMENTED** [review]
Hello and thanks for noticing this! Really appreciate it. If you are using this package it'd be nice to perform a version bump, and even test it against current `staging-next`.

File: pkgs/development/python-modules/astropy/default.nix (hunk 1 above)
Comment (LeSuisse) [reply pkgs/development/python-modules/astropy/default.nix:96]: It's not but the test failed multiple times with it on my machine. It's also quite close to some that were disabled recently for the same reason.

**Comment by LeSuisse** [comment]
> Hello and thanks for noticing this! Really appreciate it. If you are using this package it'd be nice to perform a version bump, and even test it against current `staging-next`.

I do not and the change in question has not yet been released in a tagged version upstream.

**Comment by doronbehar** [comment]
OK the version bump 6.0.1 -> 6.1.1 was done on `staging-next`. I'm really not sure about that flakey test, and I'd prefer to add it in a separate commit anyway, so I opened https://github.com/NixOS/nixpkgs/pull/323658 instead, which will also avoid merge conflicts with the [upcoming `staging-next` merge](https://github.com/NixOS/nixpkgs/pull/322145). I hope you won't mind I'll close this. Your commit was cherry-picked into #323658.
